SpatialGDK/Source/SpatialGDK/Private/Interop/Startup/DefaultServerWorkerStartupHandler.cpp

```diff
@@ -0,0 +1,459 @@
+﻿#include "Interop/Startup/DefaultServerWorkerStartupHandler.h"
+
+#include "Algo/AllOf.h"
+#include "Algo/MinElement.h"
+#include "Algo/Transform.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/GlobalStateManager.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "Schema/ServerWorker.h"
+#include "Utils/EntityFactory.h"
+#include "Utils/InterestFactory.h"
+#include "Utils/SpatialStatics.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialStartupHandler, Log, All);
+
+namespace SpatialGDK
+{
+FSpatialServerStartupHandler::FSpatialServerStartupHandler(USpatialNetDriver& InNetDriver, const FInitialSetup& InSetup)
+	: Setup(InSetup)
+	, NetDriver(&InNetDriver)
+{
+}
+
+bool FSpatialServerStartupHandler::TryFinishStartup()
+{
+	if (Stage == EStage::CreateWorkerEntity)
```

Comment:
Why is this not a switch statement?

Reply:
It's because we can execute multiple steps in a single `TryFinishStartup` call, `switch` will limit us to a single one (unless we wrap everything in a loop)

Reply:
My suggestion would be to break this out in a function with a switch and to use recursion.
```
void FSpatialServerStartupHandler::ProcessStage(const EStage Stage)
{
	switch (Stage)
	{
	case EStage::CreateWorkerEntity:
	{
		...
		if (WorkerEntityCreator->IsFinished())
		{
			...
			ProcessStage(EStage::WaitForWorkerEntities);
		}
		break;
	}
	... etc
}
```

Reply:
This will changed into individual step objects pushed to an array in a second pass, so the specific form of executing those steps does not really matter.

Reply:
Even better